packages/ui/components/ui/account-switcher.tsx

```diff
@@ -13,42 +13,85 @@ const MAX_INDEX = 2 ** 32;
 export const AccountSwitcher = ({
   account,
   onChange,
+  filter,
 }: {
   account: number;
   onChange: (account: number) => void;
+  /**
+   * An array of address indexes to switch between, if you want to limit the
+   * options. Must be pre-sorted! `AccountSwitcher` won't sort this for you.
+   */
+  filter?: number[];
 }) => {
   const [inputCharWidth, setInputCharWidth] = useState(1);
 
+  const sortedFilter = useMemo(() => (filter ? [...filter].sort() : undefined), [filter]);
+
+  const handleClickPrevious = () => {
+    if (sortedFilter) {
+      const previousAccount = sortedFilter[sortedFilter.indexOf(account) - 1];
+      if (previousAccount !== undefined) onChange(previousAccount);
+    } else onChange(account - 1);
```

Comment:
Think we should have brackets here. I was at first thought the else corresponded to the if clause above.

Reply:
Dang it, I think I forgot to self-review this. My bad. Will fix.